airflow/operators/datetime_branch_operator.py

```diff
@@ -0,0 +1,80 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import datetime
+from typing import Dict, Iterable, Optional, Union
+
+from airflow.exceptions import AirflowException
+from airflow.operators.branch_operator import BaseBranchOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class DateTimeBranchOperator(BaseBranchOperator):
+    """
+    Branches into one of two lists of tasks depending on the current datetime.
+
+    True branch will be returned when `datetime.datetime.now()` falls below
+    `target_upper` and above `target_lower`.
+
+    :param follow_task_ids_if_true: task id or task ids to follow if
+        `datetime.datetime.now()` falls above target_lower and below `target_upper`.
+    :type follow_task_ids_if_true: str or list[str]
+    :param follow_task_ids_if_false: task id or task ids to follow if
+        `datetime.datetime.now()` falls below target_lower or above `target_upper`.
+    :type follow_task_ids_if_false: str or list[str]
+    :param target_lower: target lower bound.
+    :type target_lower: Optional[datetime.datetime]
+    :param target_upper: target upper bound.
+    :type target_upper: Optional[datetime.datetime]
+    :param timezone: timezone to override local time.
+    :type timezone: Optional[datetime.timezone]
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        follow_task_ids_if_true: Union[str, Iterable[str]],
+        follow_task_ids_if_false: Union[str, Iterable[str]],
+        target_lower: Optional[datetime.datetime],
+        target_upper: Optional[datetime.datetime],
+        timezone: Optional[datetime.timezone] = None,
```

Comment:
I think this param is redundant.
We can respect the default timezone set in airflow.cfg and allow overwrite if the DAG is timezone aware.
Similar to how TimeSensor was modified https://github.com/apache/airflow/pull/9882 https://github.com/apache/airflow/pull/9699

Reply:
Thanks for pointing it out 👍 Latest commit should replicate the work done for `TimeSensor`.

Reply:
can you add unit tests ?